apps/neoscan/lib/neoscan/blocks/blocks.ex

```diff
@@ -74,14 +74,15 @@ defmodule Neoscan.Blocks do
       iex> paginate(page)
       [%Block{}, ...]
   """
-  def paginate(page) do
+  def paginate(page), do: paginate(page, @page_size)
+
+  def paginate(page, page_size) do
     block_query =
       from(
         e in Block,
         order_by: [
           desc: e.index
         ],
-        limit: @page_size,
```

Comment:
@steffenix here you removed the limit, this has a performance issue because Repo.paginate will select element and do a limit itself, but it will also run a count query on all the element to be able to return the total. In our case we have 2.5M rows + so this is not efficient. However you can bypass the count behavior by forcing total_entries:

```
Repo.paginate(block_query,
        page: page,
        page_size: @page_size,
        options: [total_entries: Counters.count_blocks()]
      )
```

After doing that you can also update this file to use the number returned by the pagination:

https://github.com/CityOfZion/neo-scan/blob/master/apps/neoscan_web/lib/neoscan_web/controllers/blocks_controller.ex